### PR TITLE
A44: gRPC xDS Retry Support

### PR DESCRIPTION
**Pull request by dapengzhang0**
@dfawley @ejona86 @markdroth Please review.

### CODE REVIEW

File: A44-xds-retry.md
Comment (markdroth) [inline A44-xds-retry.md:34]: I think this can just say "gRPC will support the following xDS retry features."  The details of why we aren't going to support individual features can be covered below.

**Review by markdroth — CHANGES_REQUESTED** [review]
This looks good overall!

My only substantive comment is the one about explaining why we are deferring or not supporting each individual feature.

Please let me know if you have any questions.  Thanks!

File: A44-xds-retry.md
Comment (ejona86) [inline A44-xds-retry.md:15]: Remove `?highlight=hedge` from URL.

File: A44-xds-retry.md
Comment (dapengzhang0) [reply A44-xds-retry.md]: Great suggestion. Thank you so much for the detailed explanation for each field!

> The retry_host_predicate and host_selection_retry_max_attempts fields, because this mechanism is different than the one described for hedging in gRFC A6, so we will evaluate this later, possibly as part of implementing hedging.

I wonder why compare these fields with hedging and possibly implement as part of hedging. Is Envoy implementing this as part of Envoy's hedging?

Comment (markdroth) [reply A44-xds-retry.md]: In gRFC A6, I believe the intent was that the feature of avoiding an endpoint that you had already sent to was primarily intended for the hedging case, because it really doesn't make sense to have two outstanding attempts of the same request to the same endpoint.  I don't think there was any particular reason why we couldn't have used the same logic for retries, but it wasn't considered as important there, which is why A6 discusses this only in the context of hedging.

If/when we actually implement something like this, I think we probably *should* do it for both retries and hedging.  But since it is more important for hedging than retries, implementing hedging will probably be the impetus for working on this feature.

File: A44-xds-retry.md
Comment (markdroth) [inline A44-xds-retry.md:44]: I think this line should be deleted.  This is already listed below under "features that we do not plan to support", and I think that's more accurate -- this is not something that we'll be adding later.

File: A44-xds-retry.md
Comment (markdroth) [inline A44-xds-retry.md:45]: I don't think we'll ever support `retry_priority`, because gRPC does not support HTTP/2 priorities.  So that belongs in the next section.

**Review by markdroth — CHANGES_REQUESTED** [review]
This looks good overall.  Just a few minor comments remaining.

File: A44-xds-retry.md
Comment (dapengzhang0) [reply A44-xds-retry.md]: Moved `retry_priority` to the next section.

**Review by markdroth — APPROVED** [review]
Looks great!

**Comment by markdroth** [comment]
(We just need to wait until Aug 16 to merge.)

**Pull request merged**
